DOC fix link to devguide triaging guide

jasonjg opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
Towards #25024


#### What does this implement/fix? Explain your changes.
The link is working in a browser. Added the link to linkcheck_ignore.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

jasonjg committed:
DOC fix link to devguide triaging guide

adrinjalali reviewed (commented):
Thanks for the pull request @jasonjg

lesteve commented on doc/conf.py:
```diff
@@ -630,6 +630,7 @@ def setup(app):
     "https://microsoft.com/",
     "https://www.jstor.org/stable/2984099",
     "https://stat.uw.edu/sites/default/files/files/reports/2000/tr371.pdf",
+    "https://devguide.python.org/triage/triaging/index.html#becoming-a-member-of-the-python-triage-team",
```
I think the issue here is actually the anchor `#becoming-a-member-of-the-python-triage-team` I have updated the issue with more info about each "broken" link.

jasonjg replied:
Thanks, will do now. I found that was the case with the Google Open source link.  My commit for that particular link will be to the corresponding .rst (about.rst)

lesteve replied:
Not sure what to do here and which section it was pointing to and what we wanted to say with this link.

I agree with @adrinjalali that we should try to find an appropriate link in this case rather than add it to the ignore list.

jasonjg committed:
update Devguide triage guide link

jasonjg committed:
Merge branch 'scikit-learn:main' into jasonjg-fix-links

adrinjalali commented on doc/governance.rst:
```diff
@@ -36,7 +36,7 @@ crucial to improve the communication in the project and limit the crowding
 
 Similarly to what has been decided in the `python project
-<https://devguide.python.org/triaging/#becoming-a-member-of-the-python-triage-team>`_,
+<https://devguide.python.org/triage/triaging/index.html>`_,
```
this sounds good to me, but the `conf.py` change then need to be reverted :)

lesteve replied:
Actually I think this may be the link we want (the anchor does still exist but in a different page somehow): https://devguide.python.org/triage/triage-team/#becoming-a-member-of-the-python-triage-team

jasonjg replied:
Working on that now. Sorry, a bit new to this so I'm trying to figure out how to properly revert that commit.  For some reason my local repo doesn't show the change I previously made.

jasonjg replied:
Working on that now.

jasonjg replied:
I removed the Devguide link from conf.py.  The pull request should be accurate and complete now.

adrinjalali replied:
@jasonjg you might have missed this comment: https://github.com/scikit-learn/scikit-learn/pull/25035#discussion_r1032174102

jasonjg replied:
> https://devguide.python.org/triage/triage-team/#becoming-a-member-of-the-python-triage-team

Just checking before I make a commit.  Is the URL I quoted the one you want to use?

lesteve replied:
Yes please :pray:.

I think this was the original intent of the link to point towards the Python Triage team policy (rather than a checklist of what to do for the Triage team)

jasonjg committed:
revert conf.py

jasonjg committed:
Update Devguide triage link

Update https://devguide.python.org/triaging/#becoming-a-member-of-the-python-triage-team to point to https://devguide.python.org/triage/triage-team/#becoming-a-member-of-the-python-triage-team

lesteve commented:
Merging, thanks a lot!
